packages/boot/test/orchestration/contract-upgrade.test.ts

```diff
@@ -0,0 +1,83 @@
+/** @file Bootstrap test of restarting contracts using orchestration */
+import { test as anyTest } from '@agoric/zoe/tools/prepare-test-env-ava.js';
+import { TestFn } from 'ava';
+
+import {
+  makeWalletFactoryContext,
+  type WalletFactoryTestContext,
+} from '../bootstrapTests/walletFactory.js';
+
+const test: TestFn<WalletFactoryTestContext> = anyTest;
+test.before(async t => {
+  t.context = await makeWalletFactoryContext(
+    t,
+    '@agoric/vm-config/decentral-itest-orchestration-config.json',
+  );
+});
+test.after.always(t => t.context.shutdown?.());
+
+/**
+ * This test core-evals a buggy installation of the sendAnywhere contract by
+ * giving it a faulty `agoricNames` service with a lookup() function returns a
+ * promise that never resolves.
+ *
+ * Because the send-anywhere flow requires a lookup(), it waits forever. This
+ * gives us a point at which we can upgrade the vat with a working agoricNames
+ * and see that the flow continues from that point.
+ */
```

Comment:
Can we clarify that the `lookup` call is not made directly in a flow, but instead from a host API which uses the `retriable` helper. As such it tests both the idempotent retry mechanism of `retriable` on upgrades, and the ability to resume an async-flow for which a host vow settles after an upgrade.